airflow/utils/log/file_task_handler.py

```diff
@@ -41,11 +41,14 @@ def __init__(self, base_log_folder, filename_template):
         self.local_base = base_log_folder
         self.filename_template = filename_template
         self.filename_jinja_template = None
+        # Should we upload our logfile when we close. Used in sub-classes, but
+        # set here and in set_context
+        self.upload_on_close = True
 
-        if "{{" in self.filename_template: #jinja mode
+        if "{{" in self.filename_template:  # jinja mode
             self.filename_jinja_template = Template(self.filename_template)
 
-    def set_context(self, ti):
+    def set_context(self, ti, upload_on_close):
```

Comment:
I'm not a big fan of this as it is not important for normal file handlers.

Reply:
That makes sense -- this is also the cause of the test failures as I didn't add the param to file_processor_handler.